syft/sbom/sbom.go

```diff
@@ -66,9 +66,11 @@ func (s SBOM) AllCoordinates() []file.Coordinates {
 	for coordinates := range s.Artifacts.Unknowns {
 		set.Add(coordinates)
 	}
-	for _, relationship := range s.Relationships {
-		for _, coordinates := range extractCoordinates(relationship) {
-			set.Add(coordinates)
+	if set.Size() > 0 {
```

Comment:
the downside with this is that we are adding relationships already from pkg-to-file or file-to-file and here we're silently dropping only the nodes, but not changing the relationships. I think this part needs a little more thought -- that is, if there is an edge in the SBOM that describes the file directly, then we either need to include that file or drop the relationship.

Reply:
hi! sounds right.. however I am not sure how to solve it.
please let me know how to proceed further so this PR can be merged in the future :)